main/streams/userspace.c

```diff
@@ -1382,6 +1384,8 @@ static int php_userstreamop_cast(php_stream *stream, int castas, void **retptr)
 	php_stream * intstream = NULL;
 	int call_result;
 	int ret = FAILURE;
+	bool report_errors = !(castas & PHP_STREAM_FLAG_SUPPRESS_ERRORS);
+	castas &= ~PHP_STREAM_FLAG_SUPPRESS_ERRORS;
```

Comment:
We could use PHP_STREAM_CAST_MASK here like in https://github.com/php/php-src/blob/ebf26af71a6e9eb800f04d1498540d187077a021/main/streams/cast.c#L193-L194

Reply:
Okay this doesn't work as ``PHP_STREAM_CAST_MASK`` does not contain the ``PHP_STREAM_FLAG_SUPPRESS_ERRORS`` flag.
```
#define PHP_STREAM_CAST_MASK		(PHP_STREAM_CAST_TRY_HARD | PHP_STREAM_CAST_RELEASE | PHP_STREAM_CAST_INTERNAL)
```

I could reuse the ``PHP_STREAM_CAST_INTERNAL`` flag but that seems kinda terrible. Not sure what to do here.

Reply:
As I mentioned a new cast flag would be better here.